infra/modules/kubevela-addons/definitions/cardano-wallet-connector.cue

```diff
@@ -0,0 +1,127 @@
+parameter: {
+	network:       *"preview" | "preprod" | "mainnet"
+}
+
+#cardanoWalletConfigs: {
+  volumes: [{
+    name: "wallet-db"
+    persistentVolumeClaim: {
+      claimName: "wallet-db"
+    }
+  }]
+  volumeMounts: [{
+    name:      "wallet-db"
+    mountPath: "/wallet-db"
+    storageClassName: "ebs-sc"
+    resources: {
+      requests: {
+        storage: "10Gi"
+      }
+    }
+  }]
+}
+
+#cardanoNodeConfigs: {
+  volumes: [{
+    name: "ipc"
+    emptyDir: {}
+  }]
+  envs: [{
+    name: "CARDANO_NODE_SOCKET_PATH"
+    value: "/ipc/node.socket"
+  }]
+}
+
+#PatchConfig: {
+  volumes: [#cardanoNodeConfigs.volumes + #cardanoWalletConfigs.volumes][0]
+  envs: [#cardanoNodeConfigs.envs][0]
+}
+
+patch: spec: {
+  // +patchStrategy=retainKeys
+  strategy: {
+    type: "Recreate"
+  }
+  template: spec: {
+    // +patchKey=name
+    volumes: #PatchConfig.volumes
+    // +patchKey=name
+    volumeMounts: #cardanoWalletConfigs.volumeMounts,
+    // +patchKey=name
+    containers: [
+      {
+        name: context.name
+        volumeMounts: [{
+          name:      #cardanoNodeConfigs.volumes[0].name
+          mountPath: "/\( #cardanoNodeConfigs.volumes[0].name )"
+        }]
+        // +patchKey=name
+        env: #PatchConfig.envs
+      },
+      {
```

Comment:
What will happen if the user also sets the `cardano-node` trait?

Reply:
this is tricky bc somehow seems that everything works flawlessly even using both of them together.

```yaml
traits:
  - type: cardano-wallet-connector
    properties:
      network: preview
  - type: cardano-node-connector
    properties:
      network: preview
```

Anyway, even with this lucky behavior I would prefer to isolate which one of them to avoid any future problems minimize the troubleshooting using the `conflictsWith` option for each other.

Reply:
I have already [added](https://github.com/input-output-hk/sc-dev-platform/pull/106/commits/e75e0bd7365b96affd509237189f15fbbf227447) those trait definitions to conflict to each other but I dont see any treatment conditions over `conflictsWith` in the source code. For sure this needs further investigation.

For now I'll add an advice at our docs since it doesnt breakes anything, sounds good?

Reply:
Sounds good

Reply:
@shlevy or since it doesnt breakes anything we could leave it as it is.

let me know what you think about it.